src/index.js

```diff
@@ -55,20 +58,28 @@ export default function attacher (remark, opts) {
                             }
                             lines[lines.length - 1].push(word);
                             len = len + word.length + 1;
-                        } else {
-                            if (pos === 0 && current.parent && is('link', current.parent)) {
-                                let paragraph = current.parent.parent.node;
-                                let index = paragraph.children.indexOf(current.parent.node);
-                                if (index > 0) {
-                                    paragraph.children[index - 1].value += `\n${indent}`;
-                                    lines[lines.length - 1].push(word);
-                                } else {
-                                    lines[lines.length - 1].push([`${indent}${word}`]);
-                                }
+                        } else if (pos === 0 && current.parent && is('link', current.parent)) {
+                            let paragraph = current.parent.parent.node;
+                            let index = paragraph.children.indexOf(current.parent.node);
+                            if (index > 0) {
+                                paragraph.children[index - 1].value += `\n${indent}`;
+                                lines[lines.length - 1].push(word);
                             } else {
-                                lines.push([`${indent}${word}`]);
+                                lines[lines.length - 1].push([`${indent}${word}`]);
                             }
                             len = word.length + 1;
+                        } else if (
+                            (len + word.length +1 >= width)
+                            && prevSibling && !is('text', prevSibling)
+                            && !(isWord.test(word) || isSpace.test(word) || !word)) {
+                            // If the current line is too long,
+                            // but the previous sibling was not text,
+                            // and the next word is not a word (e.g. punctuation),
+                            // don't create a new line for it.
```

Comment:
Will this have strange/unexpected behavior around `break` nodes? https://github.com/syntax-tree/mdast#break

Reply:
I am really not sure. I’m trusting the existing tests for this repo. If you can suggest a xxxxx.fixture.md I’ll give it a whirl and see if it breaks.